services/state/token/token.proto

```diff
@@ -0,0 +1,144 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2018 - 2023 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import "basic_types.proto";
+import "custom_fees.proto";
+
+option java_package = "com.hederahashgraph.api.proto.java";
+// <<<pbj.java_package = "com.hedera.hapi.node.state.token">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * First-draft representation of a Hedera Token Service token entity in the network Merkle tree.
+ *
+ * As with all network entities, a token has a unique entity number, which is usually given along
+ * with the network's shard and realm in the form of a shard.realm.number id.
+ */
+
+message Token {
+    /**
+     * The unique entity number of this token.
+     */
+    int64 token_number = 1;
+    /**
+     * The name of this token.
+     */
+    string name = 2;
+    /**
+     * The symbol of this token.
+     */
+    string symbol = 3;
+    /**
+     * The number of decimal places of this token.
+     */
+    int32 decimals = 4;
+    /**
+     * The total supply of this token.
+     */
+    int64 total_supply = 5;
+    /**
+     * The treasury account number of this token.
+     */
+    int64 treasury_account_number = 6;
+    /**
+     * (Optional) The admin key of this token.
+     */
+    Key admin_key = 7;
+    /**
+     * (Optional) The kyc key of this token.
+     */
+    Key kyc_key = 8;
+    /**
+     * (Optional) The freeze key of this token.
+     */
+    Key freeze_key = 9;
+    /**
+     * (Optional) The wipe key of this token.
+     */
+    Key wipe_key = 10;
+    /**
+     * (Optional) The supply key of this token.
+     */
+    Key supply_key = 11;
+    /**
+     * (Optional) The fee schedule key of this token.
+     */
+    Key fee_schedule_key = 12;
+    /**
+     * (Optional) The pause key of this token.
+     */
+    Key pause_key = 13;
```

Comment:
We should document the expectation for these keys -- like if a key is not set when the object is created, it cannot be set later, or only if there is an admin key, or whatever. Just so whoever is working in the code can see the docs right there, and not have to go back and look at the grpc objects to understand the invariants.

Reply:
Done